diff --git a/pkg/apis/orderedhttp/v1alpha1/orderedhttp_types.go b/pkg/apis/orderedhttp/v1alpha1/orderedhttp_types.go
index 2e5b185..330ad34 100644
--- a/pkg/apis/orderedhttp/v1alpha1/orderedhttp_types.go
+++ b/pkg/apis/orderedhttp/v1alpha1/orderedhttp_types.go
@@ -13,6 +13,7 @@ type OrderedHttpSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	Replicas int32 `json:"replicas"`
 }
 
 // OrderedHttpStatus defines the observed state of OrderedHttp
@@ -21,6 +22,7 @@ type OrderedHttpStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
 	// Add custom validation using kubebuilder tags: https://book.kubebuilder.io/beyond_basics/generating_crd.html
+	PodNames []string `json:"podnames"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
